contracts/contracts/multisig/Multisig.cairo

```diff
@@ -474,24 +436,113 @@ func execute_transaction{
     return (response_len=response.retdata_size, response=response.retdata)
 end
 
+# Sets number of required confirmations. The only way this can be invoked
+# is via a recursive call from execute_transaction -> set_confirmations_required.
+@external
+func set_confirmations_required{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    confirmations_required : felt
+):
+    const lower_bound = 1
+
+    let (caller) = get_caller_address()
+    let (contract_address) = get_contract_address()
+
+    with_attr error_message("caller shall be multisig"):
```

Comment:
Can we extract the `require_quorum` fn from here, and use it in both `set_confirmations_required` and `set_owners`?

Additionally, I've noticed the `require_owner` fn call is part of `submit_transaction` & `execute_transaction`. I think anyone should be able to execute the tx if M/N owners signed, and perhaps also anyone should be able to submit a transaction 🤔 

@archseer any suggestions on the above?

Reply:
> Can we extract the require_quorum fn from here, and use it in both set_confirmations_required and set_owners?
Sure, I would prefer `require_multisig/require_itself/require_caller_multisig` since here we'r not checking if quorum requirements were met and we have enough confirmations.

> anyone should be able to execute the tx if M/N owners signed, and perhaps also anyone should be able to submit a transaction
I don't see anything bad in doing this but as @archseer pointed out serum have restriction on that only owners can submit but anyone can execute it. But this varies since gnosis for example checks for owner on execution as well https://github.com/gnosis/MultiSigWallet/blob/90639984c960d281bed3e0a5d56dd4adcb9407c4/contracts/MultiSigWallet.sol#L224

So I think its up to us to decide

Reply:
`require_multisig` is ok, but that actually means M/N quorum as the call couldn't be executed any other way from the multisig :)

Yeah, let's keep proposals closed (only owners) and execution open (anyone, if quorum reached) 👍

Reply:
Done